File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        
        Map<Integer, Integer> table=new HashMap<Integer,Integer>();
        int number,result=0;
        for(int i=0;i<nums.length;i++){
            number=nums[i];
            if(table.containsKey(number)){
                table.put(number, table.get(number) + 1);
            }else{
                table.put(number,1);   
            }
            result = Collections.max(table.entrySet(), Map.Entry.comparingByValue()).getKey();
        }
        return result;
    }
}

File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count=1,currElement=nums.at(0);
        for(int i=1;i<nums.size();i++){
            if(currElement!=nums.at(i)){
                count--;
                if(count==0){
                    currElement=nums.at(i);
                    count++;
                }
            }else{
                count++;
            }
        }
        return currElement;
    }
};